### ISSUE TITLE
Create design documentation

### ISSUE THREAD

**Issue by dominic0df** [amosproj/amos2024ss08-cloud-native-llm#51]
## User story
1. As a developer
2. I want / need to create a design documentation
3. So that former developer collegues get an insight of how our software works

## Acceptance criteria
* A design documentation is created
* it contains how our architecture looks like and how our code is designed
* it replaces the existing architecture documentation
* It is published in our wiki
* PDF of wiki pages created and uploaded to deliverables folder

## Definition of done (DoD)
* Added only after week 5
* The same for all features
* Here goes the project specific part

## DoD general criteria
* Feature has been fully implemented
* Feature has been merged into the mainline
* All acceptance criteria were met
* Product owner approved features
* All tests are passing
* Developers agreed to release

**Comment by dominic0df** [comment]
added DoD
